langchain/guards/base.py

```diff
@@ -0,0 +1,78 @@
+"""Base Guard class."""
+from typing import Any, Callable, Tuple, Union
+
+
+class BaseGuard:
```

Comment:
Would suggest having this inherit from pydantic `BaseModel` -- would play more nicely with other langchain primitives that all use pydantic and would make serialization easier down the road